crates/ty_test/src/matcher.rs

```diff
@@ -297,21 +306,53 @@ impl Matcher {
                 }
             }
             ParsedAssertion::Revealed(expected_type) => {
-                #[cfg(not(debug_assertions))]
-                let expected_type = discard_todo_metadata(&expected_type);
+                let expected_type = discard_todo_metadata(expected_type);
```

Comment:
This makes sure that we have a consistent type in both debug and release modes, so that we can compare with `expected_type` down below